code/game/objects/items/devices/radio/radio.dm

```diff
@@ -503,15 +514,34 @@
 /obj/item/radio/proc/end_emp_effect(curremp)
 	if(emped != curremp) //Don't fix it if it's been EMP'd again
 		return FALSE
+	obj_flags &= ~OBJ_EMPED
 	emped = FALSE
-	set_on(TRUE)
+	set_on(previous_on)
 	return TRUE
 
 /obj/item/radio/proc/get_specific_hearers()
 	if(istype(loc, /obj/item/implant))
 		var/obj/item/implant/radio_implant = loc
 		return radio_implant.imp_in
 
+/obj/item/radio/add_strip_actions(datum/strip_context/context)
+	if (on)
+		context.add_power_off_action("The radio is on", "toggle")
+	else
+		context.add_power_on_action("The radio is off", "toggle")
+
+/obj/item/radio/perform_strip_actions(action_key, mob/actor)
+	set waitfor = FALSE
+
+	switch (action_key)
+		if ("toggle")
+			if (obj_flags & EMPED)
+				return
+			// Strip, silently
+			add_fingerprint(actor)
+			if (do_after(actor, 1 SECONDS, loc))
+				set_on(!on)
```

Comment:
Shouldn't we also add an `else` to notify the headset wearer if the do_after() fails just like with suit sensors?

Reply:
The whole point is for it to be entirely undetectable, this would defeat the purpose entirely considering how wary people already are about having someone adjacent

Reply:
I did consider adding it to the user, but I didn't want to confuse them and have them not know if the target got a message or not